File: app/src/main/java/com/walhalla/lib/Export.java
package com.walhalla.lib;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.walhalla.lib.service.Mpc;

import java.util.ArrayList;
import java.util.List;

public class Export extends BaseImages
{
    //New v2
    @SerializedName("imp")
    @Expose
    public List<String> imp = new ArrayList<>();

    @SerializedName("rxnavImageSize300")
    @Expose
    public Integer rxnavImageSize300;

    @SerializedName("nlmImageSize800")
    @Expose
    public Integer nlmImageSize800;

    @SerializedName("ingredientsAvailable")
    @Expose
    public Boolean ingredientsAvailable;
    @SerializedName("nlmImageSizeFull")
    @Expose
    public Integer nlmImageSizeFull;

    @SerializedName("acqDate")
    @Expose
    public String acqDate;
    @SerializedName("nlmImageSize600")
    @Expose
    public Integer nlmImageSize600;
    @SerializedName("rxnavImageSize120")
    @Expose
    public Integer rxnavImageSize120;
    @SerializedName("rxnavImageObjectId")
    @Expose
    public Integer rxnavImageObjectId;
    @SerializedName("deaSchedule")
    @Expose
    public String deaSchedule;
    @SerializedName("rxnavImageSize600")
    @Expose
    public Integer rxnavImageSize600;
    @SerializedName("rxnavImageSizeFull")
    @Expose
    public Integer rxnavImageSizeFull;
    @SerializedName("rxnavImageSize1024")
    @Expose
    public Integer rxnavImageSize1024;
    @SerializedName("attribution")
    @Expose
    public String attribution;
    @SerializedName("rxnavImageSize")
    @Expose
    public Integer rxnavImageSize;
    @SerializedName("nlmImageSize120")
    @Expose
    public Integer nlmImageSize120;
    @SerializedName("nlmImageSize300")
    @Expose
    public Integer nlmImageSize300;

    @SerializedName("nlmImageSize")
    @Expose
    public Integer nlmImageSize;

    @SerializedName("ingredients")
    @Expose
    public Ingredients ingredients;


    @SerializedName("nlmImageFileName")
    @Expose
    public String nlmImageFileName;

    @SerializedName("rxnavImageFileName")
    @Expose
    public String rxnavImageFileName;


    @SerializedName("_id")
    @Expose
    public Id id;
    @SerializedName("imageSize")
    @Expose
    public Integer imageSize;
    @SerializedName("rxnavImageSize800")
    @Expose
    public Integer rxnavImageSize800;


    @SerializedName("nlmImageObjectId")
    @Expose
    public Integer nlmImageObjectId;
}

File: app/src/main/java/com/walhalla/lib/Export.kt
package com.walhalla.lib

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Export : BaseImages() {
    //New v2
    @SerializedName("imp")
    @Expose
    var imp: MutableList<String?> = ArrayList<String?>()

    @SerializedName("rxnavImageSize300")
    @Expose
    var rxnavImageSize300: Int? = null

    @SerializedName("nlmImageSize800")
    @Expose
    var nlmImageSize800: Int? = null

    @SerializedName("ingredientsAvailable")
    @Expose
    var ingredientsAvailable: Boolean? = null

    @SerializedName("nlmImageSizeFull")
    @Expose
    var nlmImageSizeFull: Int? = null

    @SerializedName("acqDate")
    @Expose
    var acqDate: String? = null

    @SerializedName("nlmImageSize600")
    @Expose
    var nlmImageSize600: Int? = null

    @SerializedName("rxnavImageSize120")
    @Expose
    var rxnavImageSize120: Int? = null

    @SerializedName("rxnavImageObjectId")
    @Expose
    var rxnavImageObjectId: Int? = null

    @SerializedName("deaSchedule")
    @Expose
    var deaSchedule: String? = null

    @SerializedName("rxnavImageSize600")
    @Expose
    var rxnavImageSize600: Int? = null

    @SerializedName("rxnavImageSizeFull")
    @Expose
    var rxnavImageSizeFull: Int? = null

    @SerializedName("rxnavImageSize1024")
    @Expose
    var rxnavImageSize1024: Int? = null

    @SerializedName("attribution")
    @Expose
    var attribution: String? = null

    @SerializedName("rxnavImageSize")
    @Expose
    var rxnavImageSize: Int? = null

    @SerializedName("nlmImageSize120")
    @Expose
    var nlmImageSize120: Int? = null

    @SerializedName("nlmImageSize300")
    @Expose
    var nlmImageSize300: Int? = null

    @SerializedName("nlmImageSize")
    @Expose
    var nlmImageSize: Int? = null

    @SerializedName("ingredients")
    @Expose
    var ingredients: Ingredients? = null


    @SerializedName("nlmImageFileName")
    @Expose
    var nlmImageFileName: String? = null

    @SerializedName("rxnavImageFileName")
    @Expose
    var rxnavImageFileName: String? = null


    @JvmField
    @SerializedName("_id")
    @Expose
    var id: Id? = null

    @SerializedName("imageSize")
    @Expose
    var imageSize: Int? = null

    @SerializedName("rxnavImageSize800")
    @Expose
    var rxnavImageSize800: Int? = null


    @SerializedName("nlmImageObjectId")
    @Expose
    var nlmImageObjectId: Int? = null
}
